k8s_itests/Makefile

```diff
@@ -0,0 +1,28 @@
+# You need kind installed to run this!
```

Comment:
But then you need to install go, kubectl, and more. Those require root permission. We already have kind on devboxes, and kind is installed on .travis.yml.

Reply:
Relying on system dependencies means your code is likely to break if you run it on a different box. This is why we use tox to manage python virtualenvs, and docker to manage pretty much everything else.

On the python side, we require make and python3.6, since we [install tox in the Makefile](https://github.com/Yelp/paasta/blob/db7498cadaedbc7af8fab501fe1e74a533265e85/Makefile#L39), and use tox to install all other python dependencies.

For the mesos itests, we just require make, python3.6, and docker. Once we have docker, we install all our dependencies ourselves.

You only need root to install things if you're trying to install into root-owned directories like /usr/bin - if you set GOPATH and GOROOT to some directory within the paasta checkout, you won't need root. If you definitely need root, that's what docker is for.

Reply:
I guess there are three possible use cases. 1. devboxs. 2. travis CI. 3. others. 
1. On all devboxes, we already have the required dependencies https://sourcegraph.yelpcorp.com/sysgit/puppet@00403a1e6a221a119c5381d62dab0661b13881de/-/blob/modules/role/manifests/devbox.pp#L149-150
2. On travis CI, I installed all dependencies.
3. This is the case you talked about. And I think root is required to install go, or perhaps there is a way to workaround that I don't know? And sure I can write a script to install kubectl and kind, as on Travis. But the user still might need to configure cluster.yaml file themselves depending on their system environment. And what if the user uses Mac or Windows? What about podman? Perhaps I go too far but I initially only intend to support the first two use cases and that's why I removed the installation part.

Reply:
I guess the only solution to this problem is to containerize paasta but can we?